besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -1982,9 +1992,33 @@ private JsonRpcConfiguration createEngineJsonRpcConfiguration(
       final Integer listenPort, final List<String> allowCallsFrom) {
     JsonRpcConfiguration engineConfig =
         jsonRpcConfiguration(listenPort, Arrays.asList("ENGINE", "ETH"), allowCallsFrom);
+    engineConfig.setEnabled(isMergeEnabled());
+    if (isEngineAuthEnabled) {
+      engineConfig.setAuthenticationEnabled(true);
+      engineConfig.setAuthenticationAlgorithm(JwtAlgorithm.HS256);
+      if (engineJwtKeyFile != null && java.nio.file.Files.exists(engineJwtKeyFile)) {
+        engineConfig.setAuthenticationPublicKeyFile(engineJwtKeyFile.toFile());
+      }
+    }
     return engineConfig;
   }
 
+  private WebSocketConfiguration engineWebSocketConfiguration(
+      final Integer listenPort, final List<String> allowCallsFrom) {
+
+    final WebSocketConfiguration webSocketConfiguration =
+        webSocketConfiguration(listenPort, Arrays.asList("ENGINE", "ETH"), allowCallsFrom);
+    webSocketConfiguration.setEnabled(isMergeEnabled());
+    if (isEngineAuthEnabled) {
+      webSocketConfiguration.setAuthenticationEnabled(true);
+      webSocketConfiguration.setAuthenticationAlgorithm(JwtAlgorithm.HS256);
+      if (engineJwtKeyFile != null && java.nio.file.Files.exists(engineJwtKeyFile)) {
+        webSocketConfiguration.setAuthenticationPublicKeyFile(engineJwtKeyFile.toFile());
```

Comment:
We might want to refactor the field name here to not have public key in it

Reply:
Could also use a validation here that throws a parameter exception if a key file is supplied but engine api auth is not enabled?

Reply:
yeah, i struggled with adding another field for just symmetric keys, but i didn't want to introduce a case where a config could have multiple, and then client code may not know which to use.

Eventually engine api auth will be required. The flag to enable it will go away before the merge.